libs/core/langchain_core/prompts/chat.py

```diff
@@ -626,20 +625,31 @@ def format(self, **kwargs: Any) -> BaseMessage:
         if isinstance(self.prompt, StringPromptTemplate):
             text = self.prompt.format(**kwargs)
             return self._msg_class(
-                content=text, additional_kwargs=self.additional_kwargs
+                content=text, additional_kwargs=self.additional_kwargs, name=self.name
```

Comment:
might be worth removing the default on the name field to discover if anywhere missed in plumbing (and then re-add before merging). Feel like there are more places this should be passed through but maybe not?

Reply:
mm where else would it be outside of format/aformat?

Reply:
maybe ChatPromptTemplate.from_template?

Reply:
Thoughts? https://github.com/langchain-ai/langchain/pull/25514/commits/a11f6118a5c2b6dbfe50206190703072ff8aab7e